firestore-translate-text/functions/src/index.ts

```diff
@@ -79,13 +82,78 @@ export const fstranslate = functions.handler.firestore.document.onWrite(
     } catch (err) {
       logs.error(err);
     }
-  }
-);
+  });
+
+export const fstranslatebackfill = functions.tasks
+  .taskQueue()
+  .onDispatch(async (data: any) => {
+    const runtime = getExtensions().runtime();
+    if (!config.doBackfill) {
+      await runtime.setProcessingState(
+        "PROCESSING_COMPLETE",
+        "Existing documents were not translated because backfill was not requested."
+      );
+      return;
+    }
+    const offset = (data["offset"] as number) ?? 0;
+    const pastSuccessCount = (data["successCount"] as number) ?? 0;
+    const pastErrorCount = (data["errorCount"] as number) ?? 0;
+
+    const snapshot = await admin
+      .firestore()
+      .collection(process.env.COLLECTION_PATH)
+      .offset(offset)
+      .get();
+    const translations = await Promise.allSettled(
+      snapshot.docs.map(handleExistingDocument)
+    );
+    const newSucessCount =
+      pastSuccessCount +
+      translations.filter((p) => p.status === "fulfilled").length;
+    const newErrorCount =
+      pastErrorCount +
+      translations.filter((p) => p.status === "rejected").length;
+
+    if (snapshot.size == DOCS_PER_BACKFILL) {
+      const queue = getFunctions().taskQueue(
```

Comment:
Since this is a sample, we should consider adding some comments like "We are not done the backfill yet, enqueue the next chunk of work" or whatever.